### PR TITLE
Update tests for expo crypto to use new mocking

### PR DESCRIPTION
**Pull request by aleqsio**
# Why

We're using the new mocking mechanism for expo-crypto, tests are passing:
image

# How
Blacklisted expo-crypto from mock-generator, used `npx expo-modules-test-core generate-js-mocks` to generate mocks for it instead and bundled them in the expo module. 

# Test Plan

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Review by tsapeta — APPROVED** [review]

File: packages/expo-crypto/mocks/ExpoCrypto.ts [hunk 1]
```diff
@@ -1,10 +1,18 @@
+/**
+ *
+ * Automatically generated by expo-modules-test-core.
+ *
+ * This autogenerated file provides a mock for native Expo module,
+ * and works out of the box with the expo jest preset. */
```
Comment (tsapeta) [inline packages/expo-crypto/mocks/ExpoCrypto.ts:6]:
```suggestion
 * and works out of the box with the expo jest preset.
 */
```

File: packages/expo-crypto/mocks/ExpoCrypto.ts [hunk 2]
```diff
@@ -1,10 +1,18 @@
...
 export function digestString(algorithm, str, options) {
   return '';
 }
 export function getRandomBase64String(length) {
   return '';
 }
-export function getRandomValues(array) {}
+export function getRandomValues(array) {
+  return null; //TODO: Replace with mock for value of type TypedArray.
```
Comment (tsapeta) [inline packages/expo-crypto/mocks/ExpoCrypto.ts:14]:
```suggestion
  return null; // TODO: Replace with mock for value of type TypedArray.
```

File: packages/expo-modules-test-core/src/mockgen.ts [hunk 3]
```diff
@@ -67,6 +67,17 @@ function getMockReturnStatements(
       return [];
     case ts.SyntaxKind.ArrayType:
       return [ts.factory.createReturnStatement(ts.factory.createArrayLiteralExpression())];
+    case ts.SyntaxKind.TypeReference:
+      // can be improved by expanding a set of default mocks
+      return [
+        ts.addSyntheticTrailingComment(
+          ts.factory.createReturnStatement(ts.factory.createNull()),
+          ts.SyntaxKind.SingleLineCommentTrivia,
+          `TODO: Replace with mock for value of type ${
```
Comment (tsapeta) [inline packages/expo-modules-test-core/src/mockgen.ts:76]:
```suggestion
          ` TODO: Replace with mock for value of type ${
```

File: packages/expo-modules-test-core/src/mockgen.ts [hunk 4]
```diff
@@ -133,9 +144,19 @@ function getMockedTypes(types: Set<string>) {
   });
 }
 
+const prefix = `
+Automatically generated by expo-modules-test-core.
+
+This autogenerated file provides a mock for native Expo module,
+and works out of the box with the expo jest preset.`;
```
Comment (tsapeta) [inline packages/expo-modules-test-core/src/mockgen.ts:151]:
```suggestion
and works out of the box with the expo jest preset.
`;
```

Comment (aleqsio) [reply packages/expo-modules-test-core/src/mockgen.ts:151]: This results in:

```
 *  */
 ```

**Pull request merged**